### DISCUSSION TITLE
pytest pluggy - AssertionError pytest_runtest_logfinish

### DISCUSSION THREAD

**Discussion by lavasangee** [pytest-dev/pytest#10837] [Q&A]
`pytest` is failing with following error.
Might be related to https://github.com/pytest-dev/pluggy package

**Code**   Exception location 
https://github.com/pytest-dev/pytest/blob/main/src/_pytest/terminal.py#L598
`        assert self._session
`
    
**Error:**

    INTERNALERROR> Traceback (most recent call last):
    INTERNALERROR>   File "/home/test/project/env/lib/python3.8/site-packages/_pytest/main.py", line 270, in wrap_session
    INTERNALERROR>     session.exitstatus = doit(config, session) or 0
    INTERNALERROR>   File "/home/test/project/env/lib/python3.8/site-packages/_pytest/main.py", line 324, in _main
    INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
    INTERNALERROR>   File "/home/test/project/env/lib/python3.8/site-packages/pluggy/_hooks.py", line 265, in __call__
    INTERNALERROR>     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
    INTERNALERROR>   File "/home/test/project/env/lib/python3.8/site-packages/pluggy/_manager.py", line 80, in _hookexec
    INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
    INTERNALERROR>   File "/home/test/project/env/lib/python3.8/site-packages/pluggy/_callers.py", line 60, in _multicall
    INTERNALERROR>     return outcome.get_result()
    INTERNALERROR>   File "/home/test/project/env/lib/python3.8/site-packages/pluggy/_result.py", line 60, in get_result
    INTERNALERROR>     raise ex[1].with_traceback(ex[2])
    INTERNALERROR>   File "/home/test/project/env/lib/python3.8/site-packages/pluggy/_callers.py", line 39, in _multicall
    INTERNALERROR>     res = hook_impl.function(*args)
    INTERNALERROR>   File "/home/test/project/env/lib/python3.8/site-packages/_pytest/main.py", line 349, in pytest_runtestloop
    INTERNALERROR>     item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
    INTERNALERROR>   File "/home/test/project/env/lib/python3.8/site-packages/pluggy/_hooks.py", line 265, in __call__
    INTERNALERROR>     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
    INTERNALERROR>   File "/home/test/project/env/lib/python3.8/site-packages/pluggy/_manager.py", line 80, in _hookexec
    INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
    INTERNALERROR>   File "/home/test/project/env/lib/python3.8/site-packages/pluggy/_callers.py", line 60, in _multicall
    INTERNALERROR>     return outcome.get_result()
    INTERNALERROR>   File "/home/test/project/env/lib/python3.8/site-packages/pluggy/_result.py", line 60, in get_result
    INTERNALERROR>     raise ex[1].with_traceback(ex[2])
    INTERNALERROR>   File "/home/test/project/env/lib/python3.8/site-packages/pluggy/_callers.py", line 39, in _multicall
    INTERNALERROR>     res = hook_impl.function(*args)
    INTERNALERROR>   File "/home/test/project/env/lib/python3.8/site-packages/_pytest/runner.py", line 113, in pytest_runtest_protocol
    INTERNALERROR>     ihook.pytest_runtest_logfinish(nodeid=item.nodeid, location=item.location)
    INTERNALERROR>   File "/home/test/project/env/lib/python3.8/site-packages/pluggy/_hooks.py", line 265, in __call__
    INTERNALERROR>     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
    INTERNALERROR>   File "/home/test/project/env/lib/python3.8/site-packages/pluggy/_manager.py", line 80, in _hookexec
    INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
    INTERNALERROR>   File "/home/test/project/env/lib/python3.8/site-packages/pluggy/_callers.py", line 60, in _multicall
    INTERNALERROR>     return outcome.get_result()
    INTERNALERROR>   File "/home/test/project/env/lib/python3.8/site-packages/pluggy/_result.py", line 60, in get_result
    INTERNALERROR>     raise ex[1].with_traceback(ex[2])
    INTERNALERROR>   File "/home/test/project/env/lib/python3.8/site-packages/pluggy/_callers.py", line 39, in _multicall
    INTERNALERROR>     res = hook_impl.function(*args)
    INTERNALERROR>   File "/home/test/project/env/lib/python3.8/site-packages/_pytest/terminal.py", line 592, in pytest_runtest_logfinish
    INTERNALERROR>     assert self._session
    INTERNALERROR> AssertionError

**Comment by lavasangee** [comment]
One workaround is to add try catch. any other option to handle this cleanly?

```
try
        assert self._session
except Exception e:
        print(e)
```

**Comment by RonnyPfannschmidt** [comment]
This points to something breaking pytest

Based on what I know this assertion should never trigger in a normal pytest run
